index.bs

```diff
@@ -9971,157 +9971,109 @@ Note: The HTML Standard defines how a security check is performed. [[!HTML]]
+    and [=Realm=] |realm| is created as follows:
 
-    1.  Let |id| be the identifier of interface |I|.
-    1.  Initialize |S| to the
-        [=effective overload set=]
-        for constructors with
-        [=identifier=]
-        |id| on [=interface=]
-        |I| and with argument count 0.
-    1.  Return the length of the shortest argument list of the entries in |S|.
+    1.  Let |steps| be the following steps:
+        1.  If |I| was not declared with a [{{Constructor}}] [=extended attribute=],
+            then [=ECMAScript/throw=] a <emu-val>TypeError</emu-val>.
+        1.  If [=NewTarget=] is <emu-val>undefined</emu-val>, then
+            [=ECMAScript/throw=] a <emu-val>TypeError</emu-val>.
+        1.  Let |arg|<sub>0..|n|−1</sub> be arguments.
+        1.  Let |id| be the identifier of interface |I|.
+        1.  Initialize |S| to the [=effective overload set=]
+            for constructors with [=identifier=] |id|
+            on [=interface=] |I| and with argument count |n|.
+        1.  Let &lt;|constructor|, |values|&gt; be the result
+            of passing |S| and |arg|<sub>0..|n|−1</sub>.
+            to the [=overload resolution algorithm=].
+        1.  Let |R| be the result of performing
+            the actions listed in the description of |constructor|
+            with |values| as the argument values.
+            Rethrow any exceptions.
+        1.  Let |O| be the result of [=converted to an ECMAScript value|converting=] |R|
+            to an ECMAScript [=interface type=] value |I|.
+        1.  Assert: |O| is an object that implements |I|.
+        1.  Assert: |O|.\[[Realm]] is equal to |F|.\[[Realm]].
+        1.  Return |O|.
+    1.  Let |proto| be the [=%FunctionPrototype%=] of |realm|.
+    1.  If |I| inherits from some other interface |P|,
+        then set |proto| to the [=interface object=] of |P|.
+    1.  Let |F| be [=!=] [=CreateBuiltinFunction=](|realm|, |steps|, |proto|).
+    1.  Perform [=!=] [=SetFunctionName=](|F|, |id|).
+    1.  Let |length| be 0.
+    1.  If |I| was declared with a [{{Constructor}}] [=extended attribute=], then
+        1.  Initialize |S| to the [=effective overload set=]
+            for constructors with [=identifier=] |id| on [=interface=] |I| and
+            with argument count 0.
+        1.  Set |length| to the length of the
+            shortest argument list of the entries in |S|.
+    1.  Perform [=!=] [=DefinePropertyOrThrow=](|F|, "length",
+        PropertyDescriptor{\[[Value]]: |length|, \[[Writable]]: <emu-val>false</emu-val>, \[[Enumerable]]: <emu-val>false</emu-val>, \[[Configurable]]: <emu-val>true</emu-val>}).
+    1.  Let |obj| be the [=interface prototype object=] of [=interface=] |I|.
+    1.  Perform [=!=] [=DefinePropertyOrThrow=](|F|, "prototype",
+        PropertyDescriptor{\[[Value]]: |obj|, \[[Writable]]: <emu-val>false</emu-val>, \[[Enumerable]]: <emu-val>false</emu-val>, \[[Configurable]]: <emu-val>false</emu-val>}).
```

Comment:
That's correct, actually. |obj| is the [=interface prototype object=]. prototype. |proto| is the [[Prototype]] slot of the [=interface object=].

Agree that the variables are ill-named.